### ISSUE TITLE
[warning] can't connect to cluster node

### ISSUE THREAD

**Issue by rpavlovs** [erlio/vernemq#364]
### Environment

- VerneMQ Version: 0.15.3
- OS: 64bit Amazon Linux 2016.09

### Expected behavior

No warning in console.log

### Actual behaviour

In a two node cluster setup, on both machines, the following warning is printed to console.log every second.
Node `VerneMQ@172.31.13.31`:
```
2017-04-14 18:53:55.133 [warning] <0.599.0>@vmq_cluster_node:connect:215 can't connect to cluster node 'VerneMQ@172.31.38.151' due to econnrefused
```
Node `VerneMQ@172.31.38.151`:
```
2017-04-14 18:53:55.133 [warning] <0.599.0>@vmq_cluster_node:connect:215 can't connect to cluster node 'VerneMQ@172.31.13.31' due to econnrefused
```

I have a setup described in https://github.com/erlio/vernemq/issues/363, except all the listeners use `CONTAINER_IP_ADDRESS` instead of `AWS_HOST_IP_ADDRESS`

**Comment by dergraf** [comment]
Hi @rpavlovs 
The address/port  configured in `listener.vmq.clustering` has to be reachable from the other VerneMQ instances. The log message indicates that it cannot setup the connection.

**Comment by rpavlovs** [comment]
Hm... This means that currently there is no way to set up clustering with Docker+AWS. If I use instance host IP for `listener.vmw.clustering`, Verne errors out and exits (see linked issue).

**Comment by dergraf** [comment]
Would using the host network help in this case?

**Comment by rpavlovs** [comment]
Oh, I didn't know about the host networking mode. Will try and let you know.
Thanks

**Comment by dergraf** [comment]
Great. Host networking should also give you better networking performance, which is definitely beneficial for VerneMQ clustering.

**Comment by codeadict** [comment]
i guess using 0.0.0.0 inside the container can fix it.

**Comment by rpavlovs** [comment]
@dergraf 
I did some testing and host networking mode solves the problem. Unfortunately, ElasticBeanstalk that I was hoping to use, doesn't support it... (https://forums.aws.amazon.com/thread.jspa?threadID=177850)

I've set up ECS service, but ECS does not support exposing more that one container port to load balancer. This means that http interface can't be exposed :( (https://groups.google.com/forum/#!topic/terraform-tool/0WDwGbaDqYQ)

I'm still looking for a good Verne setup architecture on AWS...

It's interesting that, even though the warning appears in logs I'm still able to receive messages sent to nodes different than the one I'm connected to. Are there any other functionality that is broken when I see this warning?

@codeadict 
`listener.vmq.clustering = 0.0.0.0:44053` generates the following warning every second:
```
[warning] <0.1977.0>@vmq_cluster_com:loop:74 terminate due to remote_node_not_available
```
See @dergraf comment: https://github.com/erlio/vernemq/issues/364#issuecomment-294221671
